### PR TITLE
Projectile spells

### PR DESCRIPTION
**Pull request by AmosHolland**
This branch has a few things in it.

Voice recognition:
- Use a vosk model to recognise spoken word
- Use hand gestures to enable / disable voice recognition
- Do something in the game responding to voice recognition 

Projectiles:
- spawn projectiles with properties
- update projectiles based on their velocity
- detect collisions between projectiles and surfaces, and react to them

Spell casting:
- combine voice recognition and projectiles to allow for rudimentary spellcasting

This code will need to be updated to properly integrate with rollback and other systems.

### CODE REVIEW

**Review by rosefromthedead — CHANGES_REQUESTED** [review]
Other than formatting, LGTM, thanks. I'll rustfmt this and push a commit on top of what you have, please turn on "format on save" in your editors

File: src/spell_control.rs [hunk 1]
```diff
@@ -0,0 +1,214 @@
...
+            recording_mode.just_started = true;
+            recording_mode.recording = true;
+            recording_mode.just_ended = false;
+        }
+    } else if recording_mode.recording {
+        recording_mode.just_ended = true;
+        
+    }
+}
+*/
```
[the remark this answers is not in the document]
Comment (AmosHolland) [reply src/spell_control.rs]: It's dead, now removed :+1:

**Comment by AmosHolland** [comment]
This is working in single player, but we haven't had success with multiplayer yet. If we need to push this through for the sake of the demo it will work for that.

**Review by OwenBoisse — CHANGES_REQUESTED** [review]
we now have a zipped vosk-model in order to bake the model into the apk, therefore we need to remove the original vosk-model folder

**Review by Foxinatel — CHANGES_REQUESTED** [review]
- Remove `build.rs` if it's unnecessary
- Remove the file called `grep` which contains the output of `iw --help`
- Move `/src/vosk-model.zip` to a directory that isn't `/src`
- Remove unused function `spawn_vr_camera` and struct `VRCamera`
- Clean up some remaining cargo clippy lints

**Comment by LICHENGWEI-cloud** [comment]
the visual effect works well for me. When I say the colour word and hold the exact gesture. the ball corresponding to that colour is generated on my right hand correctly. And the ball will be emitted when I hold the gesture corresponding to the emission action.

**Comment by AmosHolland** [comment]
> * Remove `build.rs` if it's unnecessary
> 
>     * Remove the file called `grep` which contains the output of `iw --help`
> 
>     * Move `/src/vosk-model.zip` to a directory that isn't `/src`
> 
>     * Remove unused function `spawn_vr_camera` and struct `VRCamera`
> 
>     * Clean up some remaining cargo clippy lints

I believe these are now done. Some linting issues are still there, but these are about unused imports that are actually used when we run on the headset.

**Comment by AmosHolland** [comment]
> We now have a zipped vosk-model in order to bake the model into the apk, therefore we need to remove the original vosk-model folder.

This is done now.

**Review by YiwenLiang-0-0 — APPROVED** [review]
Voice recognition and spells projectile works well. In the test, can control whether to recognize voice based on gestures, generate balls of different colors based on different spells, and launch them through gestures.

**Review by OwenBoisse — APPROVED** [review]
Looks all good 👍

**Review by rosefromthedead — APPROVED** [review]
Get that shit merged

**Review by Foxinatel — APPROVED** [review]
I support the LGTM community and approve their changes

**Pull request merged**